llvm/lib/Target/AArch64/GISel/AArch64LegalizerInfo.cpp

```diff
@@ -1070,6 +1070,13 @@ AArch64LegalizerInfo::AArch64LegalizerInfo(const AArch64Subtarget &ST)
                  {s16, v8s16},
                  {s32, v2s32},
                  {s32, v4s32}})
+      .moreElementsIf(
```

Comment:
I think this can happen for more than just odd numbers, if we have support in the legalizer. I think I would make it moreElementsToNextPow2 unless there is a big reason not to.

Reply:
As I noted in my commit message, unfortunately that causes regressions for odd-sized `i8` vectors: https://gist.github.com/dc03-work/3d749a7be0dc893d86d2df0fbc31709a (except for the very last case... for some reason). I was noticing another test failure when I enabled it for even-sized vectors, however that seems to have gone away now.